### PR TITLE
Propagate attributes when resharing

### PR DESCRIPTION
**Pull request by CarlSchwan**
When updating a share, load the node from the initiator instead of the owner similar to how this is done when creating the share.

### CODE REVIEW

File: lib/private/Share20/Share.php [hunk 1]
```diff
@@ -181,13 +181,7 @@ public function getNode() {
 				throw new NotFoundException();
 			}
 
-			// for federated shares the owner can be a remote user, in this
-			// case we use the initiator
```
[the remark this answers is not in the document]
Comment (CarlSchwan) [reply lib/private/Share20/Share.php]: I just changed the way to get the reshared node. This doesn't touch this logic anymore but semi copy it in checkInheritedAttributes

File: apps/files_sharing/lib/Controller/ShareAPIController.php [hunk 2]
```diff
@@ -1926,6 +1914,9 @@
 			$inheritedAttributes = $storage->getShare()->getAttributes();
 			if ($inheritedAttributes !== null && $inheritedAttributes->getAttribute('permissions', 'download') === false) {
 				$share->setHideDownload(true);
+				$attributes = $share->getAttributes();
+				$attributes->setAttribute('permissions', 'download', false);
```
[the remark this answers is not in the document]
Comment (PVince81) [reply apps/files_sharing/lib/Controller/ShareAPIController.php:1918]: please double check this, am not sure if you need to instantiate in some cases

**Review by PVince81 — APPROVED** [review]
:+1: see warning

**Review by skjnldsv — APPROVED** [review]

**Comment by CarlSchwan** [comment]
Ci failure is unrelated: I double checked an  /drone/src/tests/acceptance/features/app-files-sharing.feature:402 is also failling on other commits on master

**Comment by PVince81** [comment]
bot, where is my backport ?

I'll do it myself

**Comment by PVince81** [comment]
stable24: https://github.com/nextcloud/server/pull/34840

**Pull request merged**